addons/misra.py

```diff
@@ -1494,6 +1494,254 @@ def misra_8_14(self, rawTokens):
+                return True
+
+            # Initializer must start with a curly bracket
+            if not token.str == '{':
+                self.reportError(token, 9, 2)
+                return False
+
+            # Empty initializer is not ok { }
+            if not token.astOperand1:
+                self.reportError(token, 9, 2)
+                return False
+
+            token = token.astOperand1
+
+            # Zero initializer is ok { 0 }
+            if token.str == '0' :
+                return True
+
+            pos = None
+            while(token):
+                if token.str == ',':
+                    token = token.astOperand1
+                else:
+                    if pos == None:
+                        pos = 0
+
+                    if token.isAssignmentOp:
+                        if token.astOperand1.str == '.':
+                            elementName = token.astOperand1.astOperand1.str
+                            pos = next((i for i, element in enumerate(elements) if element.name == elementName), len(elements))
+                        token = token.astOperand2
+
+                    if pos >= len(elements):
+                        self.reportError(token, 9, 2)
+                        return False
+
+                    element = elements[pos]
+                    if element.elementType == 'class':
+                        if token.isName:
+                            if not token.valueType.typeScope  == element.valueType.typeScope:
+                                self.reportError(token, 9, 2)
+                                return False
+                        else:
+                            subElements = getRecordElements(element.valueType)
+                            if not checkObjectInitializer(token, subElements):
+                                return False
+                    elif element.elementType == 'array':
+                        if not checkArrayInitializer(token, element.dimensions, element.valueType):
+                            return False
+                    elif token.str == '{':
+                        self.reportError(token, 9, 2)
+                        return False
+
+                    # The assignment represents the astOperand
+                    if token.astParent.isAssignmentOp:
+                        token = token.astParent
+
+                    if not token == token.astParent.astOperand2:
+                        pos = pos + 1
+                        token = token.astParent.astOperand2
+                    else:
+                        token = None
+
+            return True
+
+        # ------
+        for token in data.tokenlist:
+            if token.variable:
+                variable = token.variable
+                nameToken = variable.nameToken
+                # Find declarations (also if isSplittedVarDeclEq is True)
+                if nameToken.file == token.file and nameToken.linenr == token.linenr and nameToken.column == token.column:
```

Comment:
I suggest `nameToken == token` or `nameToken.Id == token.Id`.

Reply:
We needed to find the nameToken used in the initialization assignment when the declaration and assignment was splitted in the ast, as we've seen happen for structs and unions. I've changed this now to.

    if nameToken.next.isSplittedVarDeclEq:
        nameToken = nameToken.next.next
    
    if nameToken is token:

Can we rely on this approach to always produce the desired result? It seems correct from what I can see from the dump file.

Reply:
Yes I believe we can rely on that approach.